common-lib/src/main/scala/com/gu/mediaservice/lib/auth/PermissionsHandler.scala

```diff
@@ -0,0 +1,30 @@
+package com.gu.mediaservice.lib.auth
+
+import scala.concurrent.{ExecutionContext, Future}
+
+
+object PermissionDeniedError extends Throwable("Permission denied")
+
+trait PermissionsHandler {
+  val failFuture = Future.failed(PermissionDeniedError)
+  val permissionStore: PermissionStore
+
+  def validateUserWithPermissions(user: Principal, permission: PermissionType.PermissionType)
+                                 (implicit ec: ExecutionContext): Future[Principal] =
```

Comment:
There's two different methods here which are trying to express solutions different situations:

__[`validateUserWithPermissions`](https://github.com/guardian/grid/pull/1163/files#diff-3047d82dcca27850c68ce76fae73e54cR12)__
This is when you want to do something once we know the user is validated to do it, e.g. delete crops. This is similarly to our Authenticated thing. [I was playing with the idea of using Scalaz`s Validations](https://github.com/guardian/grid/commit/4df32cd6fbc46d295383c04efd9145623ee3c8ef#diff-f1869cbcbc4bbc5ee4ddb9c12741f6e9R133) -> but felt it overkill where promises can express the same thing.

__[getPermissionValForUser](https://github.com/guardian/grid/pull/1163/files#diff-3047d82dcca27850c68ce76fae73e54cR19)__
When you want the value of the permission.

> Future[Boolean]
Not sure it's massively useful, but then again, neither is the user.